GitUI/BranchTreePanel/LocalBranchNode.cs

```diff
@@ -76,5 +81,27 @@ public bool Rename()
         {
             return UICommands.StartRenameDialog(ParentWindow(), branch: FullPath);
         }
+
+        public void UpdateAheadBehind(string aheadBehindData, string relatedBranch)
+        {
+            AheadBehind = aheadBehindData;
+            RelatedBranch = relatedBranch;
+        }
+
+        protected override string DisplayText()
+        {
+            return string.IsNullOrEmpty(AheadBehind) ? Name : $"{Name} ({AheadBehind})";
+        }
+
+        protected override void SelectRevision()
+        {
+            TreeViewNode.TreeView?.BeginInvoke(new Action(() =>
+            {
+                string branch = RelatedBranch is null || !RepoObjectsTree.ModifierKeys.HasFlag(Keys.Alt)
+                    ? FullPath : RelatedBranch.Substring(startIndex: GitRefName.RefsRemotesPrefix.Length);
```

Comment:
This used to work for remote branches, too...

Reply:
Related branch was not set only for Local branch. I was thinking about this myself.
Not tonight, probably separate PR

Reply:
OK

Reply:
I doubt that this has ever worked.
The solution should also eliminate the additional Git command needed for push button vs left panel (as remote/local should share the info anyway). The aheadBehindProvider should be made lazy or the info be provided as the refs. Lazy data reset when the refs are refreshed.

Slightly offtopic, but should remote branches be decorated in some way?
The easiest is to have exactly the same info as locals. The info could be prepended with "local: " or so but that is annoying when you know about it. Tooltip could add details.